src/main/java/mate/academy/dao/impl/AbstractDao.java

```diff
@@ -0,0 +1,56 @@
+package mate.academy.dao.impl;
+
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+public abstract class AbstractDao {
+    protected final SessionFactory factory;
+
+    public AbstractDao(SessionFactory factory) {
+        this.factory = factory;
+    }
+
+    protected <T> T create(T entity) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(entity);
+            transaction.commit();
+            return entity;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add a "
+                    + entity.getClass().getSimpleName() + " = [" + entity + "]", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    protected <T> Optional<T> get(Class<T> entityClass, Long id) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```

Comment:
Here again, you should use the `factory` field to open the session instead of `HibernateUtil.getSessionFactory().openSession()` for the same reasons mentioned above.